File: problems/338-counting-bits.py
class Solution:
    def get_set_bits_count(self, num):
        count = 0
        while num:
            count += 1
            num &= num - 1
        return count

    def countBits(self, num):
        result = []
        for n in range(num + 1):
            result.append(self.get_set_bits_count(n))
        return result

File: leetcode/java/338CountingBits.java
public class Solution {
  public int[] countBits(int num) {
      int[] bitCounts = new int[num + 1];
      for (int i = 1; i <= num; ++i) {
		int withoutLastSetBit = bitCounts[i & (i - 1)];
		bitCounts[i] = withoutLastSetBit + 1;
	}
      return bitCounts;
  }
}
